src/Login/Login.test.js

```diff
@@ -14,28 +21,39 @@ describe('Login', () => {
     expect(screen.getByPlaceholderText('Email')).toBeInTheDocument()
     expect(screen.getByText('Login')).toBeInTheDocument()
   })
+
   it('should recieve an error if one of the inputs is empty', () => {
     render(<Login />)
     userEvent.type(screen.getByPlaceholderText('Email'), 'test@turing.io')
     userEvent.click(screen.getByText('Login'))
     expect(screen.getByText('Please type your email and password to sign in')).toBeInTheDocument()
   })
-  it('should submit the login information when clicked', () => {
-    const fakeSubmit = jest.fn();
-    const fakeUser = {user: {id: 88, name: 'Olivia', email: 'olivia@turing.io'}}
-    //render login
-    render(
-      <Login
-        {...fakeUser}
-        handleSubmit={fakeSubmit}
-      />
+
+  it('should welcome the user after submitting login info', async() => {
+    ApiCalls.postUserLogin.mockResolvedValueOnce(
+      {user: {id: 888, name: 'Olivia', email: 'olivia@turing.io'} }
     )
-    //enter user info
-    userEvent.type(screen.getByPlaceholderText('Email'), 'test@turing.io')
+
+    render(<BrowserRouter><App /></BrowserRouter>)
+    expect(screen.getByText('Login!')).toBeInTheDocument()
+
+
+    userEvent.click(screen.getByText('Login!'))
+    expect(screen.getByPlaceholderText('Email')).toBeInTheDocument()
+
+    userEvent.type(screen.getByPlaceholderText('Email'), 'olivia@turing.io')
     userEvent.type(screen.getByPlaceholderText('Password'), 'pword')
-    // userEvent.click(screen.getByText('Login'))
-    // expect(fakeSubmit).toHaveBeenCalledTimes(1);
-    //click the button
-    //assert that submit login was called
+    userEvent.click(screen.getByText('Login'))
+
+    const welcomeMessage = await waitFor(() => screen.getByText('Welcome, Olivia!'))
+    expect(welcomeMessage).toBeInTheDocument();
+    ApiCalls.getAllMovies.mockResolvedValueOnce(
```

Comment:
I think we might have this mock functionality happening at the wrong time in this test. After speaking with my mentor I think we should have this happening at the top of our it block.